### PR TITLE
Use channelz to determine the ADS connections state 

### PR DESCRIPTION
**Pull request by eugeneo**
This way we can detect when the client starts talking to an expected control plane. This makes the test more stable.

[Test run](https://fusion2.corp.google.com/ci/kokoro/prod:grpc%2Fcore%2Fmaster%2Flinux%2Fpsm-fallback/activity/9ddb32fc-899a-404b-9c17-70ee774d736b/log)

### CODE REVIEW

File: framework/helpers/docker.py [hunk 1]
```diff
@@ -318,8 +265,16 @@ def __init__(
...
             image=image,
             name=name,
-            command=[f"--server={url}", "--print_response"],
-            ports={DEFAULT_GRPC_CLIENT_PORT: port},
+            command=[
+                "--server",
+                url,
+                # "--print_response",
+                # "true",
+                # "--verbose",
```
Comment (XuanWang-Amos) [inline framework/helpers/docker.py:273]: Should we delete those?

File: framework/helpers/docker.py [hunk 2]
```diff
@@ -333,7 +288,28 @@ def get_stats(self, num_rpcs: int):
...
+        channelz = ChannelzServiceClient(self.channel())
+        status = None
+        while datetime.datetime.now() < deadline:
+            status = None
+            for ch in channelz.list_channels():
+                if ch.data.target.endswith(str(port)):
+                    status = ch.data.state.state
+                    break
+            if status == expected_status:
+                return status
```
Comment (XuanWang-Amos) [inline framework/helpers/docker.py:313]: We're already checking status use `assertEqual`, should we consider just return a status here directly?

**Review by XuanWang-Amos — COMMENTED** [review]
Please add a test run to description.

**Review by eugeneo — COMMENTED** [review]
Thank you for the review. I added a link to a successful run (with your comments addressed) into the PR description.

**Review by XuanWang-Amos — APPROVED** [review]

**Pull request merged**